tests/test_discovery.py

```diff
@@ -0,0 +1,99 @@
+                                     if catalog["stream_name"] == stream]))
+                self.assertIsNotNone(catalog)
+
+                # collecting expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_replication_method = self.expected_replication_method()[stream]
+
+                # add primary keys and replication keys in automatically replicated keys to check
+                expected_automatic_fields = expected_primary_keys | expected_replication_keys
+
+                # collecting actual values...
+                schema_and_metadata = menagerie.get_annotated_schema(conn_id, catalog['stream_id'])
+                metadata = schema_and_metadata["metadata"]
+                stream_properties = [item for item in metadata if item.get("breadcrumb") == []]
+                actual_primary_keys = set(stream_properties[0].get("metadata", {self.PRIMARY_KEYS: []}).get(self.PRIMARY_KEYS, []))
+                actual_replication_keys = set(stream_properties[0].get("metadata", {self.REPLICATION_KEYS: []}).get(self.REPLICATION_KEYS, []))
+                actual_automatic_fields = set(item.get("breadcrumb", ["properties", None])[1] for item in metadata
+                                              if item.get("metadata").get("inclusion") == "automatic")
+                actual_replication_method = stream_properties[0].get("metadata", {self.REPLICATION_METHOD: None}).get(self.REPLICATION_METHOD)
+
+                ##########################################################################
+                ### metadata assertions
+                ##########################################################################
+
+                # verify there is only 1 top level breadcrumb in metadata
+                self.assertTrue(len(stream_properties) == 1,
+                                msg="There is NOT only one top level breadcrumb for {}".format(stream) + \
+                                "\nstream_properties | {}".format(stream_properties))
+
+                # verify primary key(s) match expectations
+                self.assertSetEqual(expected_primary_keys, actual_primary_keys,)
+
+                # verify replication key(s) match expectations
+                self.assertSetEqual(expected_replication_keys, actual_replication_keys,)
+
+                # verify that if there is a replication key we are doing INCREMENTAL otherwise FULL
+                if expected_replication_keys:
+                    self.assertEqual(self.INCREMENTAL, actual_replication_method)
+                else:
+                    self.assertEqual(self.FULL_TABLE, actual_replication_method)
+
+                # verify the replication method matches our expectations
+                self.assertEqual(expected_replication_method, actual_replication_method)
+
```

Comment:
Could you please add an assertion to verify the inclusion metadata at the stream level. I believe it should be at `stream_properties[0]['metadata']['inclusion']`. It should be equal to "available". This was an assertion that was recently added to this base test by a new QA on our team. We will update this in the tap-tester expectations for saas taps.

Reply:
@kspeer825 There is not **inclusion** in top level breadcrumb in the tap. Here is the sample top level metadata for reference `{
    "breadcrumb": [],
    "metadata": {
        "table-key-properties": [
            "gid"
        ],
        "forced-replication-method": "FULL_TABLE"
    }
}`